YouTube.js

```diff
@@ -90,15 +90,14 @@ function doWeb(doc, url) {
 function scrape(doc, url) {
 	var item = new Zotero.Item("videoRecording");
 	item.title = text(doc, '#info-contents h1.title');
-	item.url = url;
+	item.url = ZU.xpathText(doc, '//link[@rel="canonical"]/@href') || url;
 	item.runningTime = text(doc, '#movie_player .ytp-time-duration');
 	
-	item.date = text(doc, '#info-text #date');
+	item.date = ZU.xpathText(doc, '//meta[@itemprop="datePublished"]/@content');
 	if (item.date) {
 		item.date = ZU.strToISO(item.date);
 	}
-	var author = text(doc, '#meta-contents #text-container .ytd-channel-name')
-		|| text(doc, '#text-container .ytd-channel-name');
+	var author = text(doc, '#text-container.ytd-channel-name > yt-formatted-string > a');
```

Comment:
If we're matching on an ID, we don't need to match on a class, too, do we? Will there ever ben a `#text-container` without the `.ytd-channel-name` class that we want to exclude?

Also, is this broken right now? It's working fine on my end, and the `#meta-contents` tweak was to prevent the wrong creator name from being scraped, so it should stay unless it's causing issues.

Reply:
I'll revert it.
It's working, I guess confused it with another error and didn't notice that the nested element also has the "ytd-channel-name" class while testing.